deepinv/optim/prior.py

```diff
@@ -0,0 +1,170 @@
+import torch
+import torch.nn as nn
+
+from deepinv.optim.utils import gradient_descent
+
+
+class Prior(nn.Module):
+    r"""
+    Prior term :math:`g{x}`. To implement a custom prior, for an explciit prior, overwrite `g` (do not forget to specify `self.explicit_prior = True`). For an implicit prior, overwrite the `grad` or `prox`.
+
+    This is the base class for the prior term :math:`g{x}`.
+    """
+
+    def __init__(self, g=None, grad=None, prox=None):
+        super().__init__()
+        self._g = g
+        self._grad = grad
+        self._prox = prox
+        self.explicit_prior = False if self._g is None else True
+
+    def g(self, x, *args, **kwargs):
```

Comment:
why double fundction g and forward?

Reply:
@samuro95 so we are leaving this as is right?